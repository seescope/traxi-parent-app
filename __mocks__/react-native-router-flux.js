import React, { PropTypes } from 'react';
import { View } from 'react-native';

export const Actions = {
  createKid: jest.fn(),
  walkthrough: jest.fn(),
  congratulations: jest.fn(),
  thankyou: jest.fn(),
  areYouReady: jest.fn(),
  intro: jest.fn(),
};

export const Scene = (props) => <View {...props} />;

export const Router = ({ children }) => (
  <View>{children}</View>
);

Router.propTypes = {
  children: PropTypes.array.isRequired,
};
