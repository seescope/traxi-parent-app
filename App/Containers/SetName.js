import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Keyboard, Text, View } from 'react-native';
import { Actions } from 'react-native-router-flux';

import { enterKidName } from '../Actions/Actions';
import Button from '../Components/Button';
import TextInput from '../Components/TextInput';
import HeaderText from '../Components/HeaderText';
import Spacing from '../Components/Spacing';
import { WHITE, TRANSPARENT } from '../Constants/Colours';

const style = {
  bodyText: {
    fontFamily: 'Raleway-Regular',
    fontSize: 16,
    color: WHITE,
    backgroundColor: TRANSPARENT,
    textAlign: 'left',
  },
  innerContainer: {
    flex: 4,
  },
  buttonContainer: {
    alignItems: 'center',
  },
};

export const setKidName = kidName => dispatch => dispatch(enterKidName(kidName));

const nextStep = () => {
  Keyboard.dismiss();
  Actions.setImage();
};

const SetName = ({ onChangeText }) => (
  <View style={style.container}>
    <HeaderText>Let's get started!</HeaderText>

    <Spacing height={32} />

    <Text style={style.bodyText}>
      What is your kid's name?
    </Text>

    <Spacing height={16} />

    <TextInput
      refFunc={ref => { this.textInput = ref; }}
      onChangeText={onChangeText}
      onSubmitEditing={() => Actions.setImage()}
    />

    <Spacing />

    <View style={style.buttonContainer}>
      <Button onPress={() => nextStep()}>Next step</Button>
    </View>
  </View>
);

SetName.propTypes = {
  parentName: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  parentName: state.parentName,
});

const mapDispatchToProps = {
  onChangeText: setKidName,
};

export default connect(mapStateToProps, mapDispatchToProps)(SetName);
