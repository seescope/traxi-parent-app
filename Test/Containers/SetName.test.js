import React from 'react';
import renderer from 'react-test-renderer';
import { Keyboard, Alert } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
jest.mock('../../App/AsyncActions/PersistKid', () =>
  jest.fn(() => Promise.resolve({ type: 'Hey' })));
jest.mock('../../App/AsyncActions/WatchKid', () =>
  jest.fn(() => Promise.resolve({ type: 'Hey' })));
import mockPersistKid from '../../App/AsyncActions/PersistKid';
import mockWatchKid from '../../App/AsyncActions/WatchKid';

import SetName, { mergeProps, verifyName } from '../../App/Containers/SetName';

const mockStore = configureStore([thunk]);
const testStore = mockStore({
  kidsState: {
    'abc-123': {
      name: 'John Bobson',
    },
  },
  setupState: {
    kidUUID: 'abc-123',
  },
});

const SetNameComponent = () => (
  <Provider store={testStore}>
    <SetName />
  </Provider>
);

it('renders the <SetName> component', () => {
  const tree = renderer.create(<SetNameComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('handles setting the kid\'s name', () => {
  const mockDispatch = jest.fn();

  const { onNameChanged } = mergeProps(
    { kidUUID: 'abc-123' },
    { dispatch: mockDispatch },
  );
  onNameChanged('TEST_NAME');
  expect(mockDispatch).toHaveBeenCalledWith({
    name: 'TEST_NAME',
    UUID: 'abc-123',
    type: 'SET_KID_NAME',
  });
});

it('verifies the kid\'s name was input correctly', () => {
  Keyboard.dismiss = jest.fn();
  Alert.alert = jest.fn();

  verifyName('hey');
  expect(Keyboard.dismiss).toHaveBeenCalled();
  expect(Alert.alert).not.toHaveBeenCalled();

  Keyboard.dismiss.mockClear();
  Alert.alert.mockClear();

  verifyName(' ');

  expect(Keyboard.dismiss).toHaveBeenCalled();
  expect(Alert.alert).toHaveBeenCalled();
});

it('persists the kid and navigates to the next screen onPress', () => {
  const mockDispatch = jest.fn(v => v);
  const TEST_KID_NAME = 'Jim';

  const { onPress } = mergeProps(
    { kidName: TEST_KID_NAME },
    { dispatch: mockDispatch },
  );

  return onPress(TEST_KID_NAME).then(() => {
    expect(Actions.deviceSetup).toHaveBeenCalled();
    expect(mockPersistKid).toHaveBeenCalled();
    expect(mockWatchKid).toHaveBeenCalled();
  });
});
