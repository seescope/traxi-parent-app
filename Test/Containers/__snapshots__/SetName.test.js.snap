// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles going to the nextStep 1`] = `
Array [
  Array [
    Object {
      "type": "NEXT_STEP",
    },
  ],
]
`;

exports[`handles setting the kid's name 1`] = `
Array [
  Array [
    Object {
      "kidName": "Test",
      "type": "ENTER_KID_NAME",
    },
  ],
]
`;

exports[`renders the <SetName> component 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 30,
        "marginBottom": -2,
        "textAlign": "center",
      }
    }
  >
    [missing "en.setName.header" translation]
  </Text>
  <View
    style={
      Object {
        "height": 32,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 16,
        "textAlign": "left",
      }
    }
  >
    [missing "en.setName.kidsName" translation]
  </Text>
  <View
    style={
      Object {
        "height": 16,
      }
    }
  />
  <TextInput
    autoCapitalize="words"
    autoCorrect={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="rgba(75, 73, 73, 0.8)"
    style={
      Object {
        "backgroundColor": "#EFEFEF",
        "borderRadius": 4,
        "color": "rgba(75, 73, 73, 1)",
        "fontFamily": "Raleway-Regular",
        "height": 32,
        "paddingHorizontal": 8,
        "width": 686,
      }
    }
    underlineColorAndroid="rgba(255, 255, 255, 1)"
  />
  <View
    style={
      Object {
        "height": 24,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }
      >
        [missing "en.general.nextStep" translation]
         
      </Text>
    </View>
  </View>
</View>
`;
